ddtrace/llmobs/_llmobs.py

```diff
@@ -74,52 +76,89 @@ def _stop_service(self) -> None:
         except Exception:
             log.warning("Failed to shutdown tracer", exc_info=True)
 
+    @classmethod
+    def _start_llmobs_writers(cls):
+        try:
+            cls._instance._llmobs_span_writer.start()
+            cls._instance._llmobs_eval_metric_writer.start()
+        except ServiceStatusError:
+            log.debug("LLMObs writers already started")
+
+    @classmethod
+    def _stop_llmobs_writers(cls):
+        try:
+            cls._instance._llmobs_span_writer.stop()
+            cls._instance._llmobs_eval_metric_writer.stop()
+        except ServiceStatusError:
+            log.debug("LLMObs writers already stopped")
+
     @classmethod
     def enable(cls, tracer=None):
-        if cls._instance is not None:
+        if cls.enabled:
             log.debug("%s already enabled", cls.__name__)
             return
 
+        if os.getenv("DD_LLMOBS_ENABLED") and not asbool(os.getenv("DD_LLMOBS_ENABLED")):
+            cls.enabled = False
+            config._llmobs_enabled = False
+            log.debug("LLMObs.enable() called when DD_LLMOBS_ENABLED is set to false or 0, not starting LLMObs service")
+            return
+
         if not config._dd_api_key:
             cls.enabled = False
+            config._llmobs_enabled = False
             raise ValueError(
                 "DD_API_KEY is required for sending LLMObs data. "
                 "Ensure this configuration is set before running your application."
             )
         if not config._llmobs_ml_app:
             cls.enabled = False
+            config._llmobs_enabled = False
             raise ValueError(
                 "DD_LLMOBS_APP_NAME is required for sending LLMObs data. "
                 "Ensure this configuration is set before running your application."
             )
-
+        # override the default _instance with a new tracer
         cls._instance = cls(tracer=tracer)
+
+        # flip on llmobs enabled booleans
         cls.enabled = True
+        config._llmobs_enabled = True
```

Comment:
`is_pc_sampled_llmobs` looks at `config._llmobs_enabled` to check whether or not to tag spans with llmobs data, so we need to make sure `config._llmobs_enabled` is true whenever we add the LLMObs trace processor

otherwise, for example,
1. bedrock makes a span
2. config._llmobs_enabled=False, so the integration doesn't add any llmobs data
3. the span is marked as type LLM no matter what
4. the span get's picked up by the LLMObs trace processor
5. we try to process a span without LLMObs data, and this throws an error

Reply:
I would prefer to keep `config._llmobs_enabled` a static value that we don't toggle, and instead change `is_pc_sampled_llmobs` to check `LLMObs.enabled` directly. Thoughts?

Reply:
Agree with that